### ISSUE TITLE
[rush] filter specific warnings during CI

### ISSUE THREAD

**Issue by robertf224** [microsoft/rushstack#3938]
Just started using the new Next.js [app directory](https://beta.nextjs.org/docs/upgrade-guide#migrating-from-pages-to-app) in our Rush repo.  The feature is listed as "experimental" in Next.js, and they emit this warning during builds:
<img width="595" alt="Screen Shot 2023-01-29 at 8 25 42 PM" src="https://user-images.githubusercontent.com/671815/215369034-af657e09-eac0-4454-84fd-a4fbb3cc7dac.png">

Because of the warning, our Rush CI build now starts to fail.  There's an [upstream issue](https://github.com/vercel/next.js/discussions/32192) tracking this, but obviously not predictable when it'll be addressed.  We obviously *could* use Rush's `allowWarningsInSuccessfulBuild` as an escape hatch here, but I'd rather not sacrifice the integrity of our entire job based on one warning in one dependency.

It would be cool if there was a way to filter out / allow specific warnings during the job.  There's obviously a balance here because the reason the setting exists in the first place is to prevent developers from ignoring warnings, but there is a difference between warnings you're fine with / have accepted risks for (i.e. this one in our case) and warnings you're seeing for the first time which should actually fail your job.  The practical alternative of turning off warnings for the entire repo seems worse here, and assuming there were some configuration file for allowed warnings, a team could use CODEOWNERS to add technical controls against accepting every warning, and could leave comments that explain the reason for filtering the warning and tracking upstream issues.

Intuitively seems like something that could/should be delivered as a Rush plugin (related in spirit to [this open Heft plugin PR](https://github.com/microsoft/rushstack/pull/3866)).  This won't actually work (couldn't find a real way to plug into the `Terminal` for a Rush session on first glance) but would imagine code looking something like this:
<img width="790" alt="Screen Shot 2023-01-29 at 8 23 57 PM" src="https://user-images.githubusercontent.com/671815/215370222-dfcf4a27-caa7-4c02-9901-59fd1891de14.png">

**Comment by iclanton** [comment]
Yeah ideally you could suppress that specific warning with an environment variable or some other flag.

This does seem like a useful feature. This could go into a plugin (we've been expanding the plugin API surface as people have written plugins that need more APIs), or it could go into the `rush-project.json` file.

Would you be interested in putting this together?

**Comment by robertf224** [comment]
yeah I'm happy to build it + can put together a proposal.  could use guidance here on a gut check around shipping it as a plugin or as a core feature -- if it's core it's obviously easier to manipulate the terminal output, whereas if it's a plugin (something like my pseudo-code above) we'll need some generic way for plugins to add some sort of processor to a stack of terminal output processors (which probably requires some changes to the existing `Terminal` API).  plugin route might not be worth the squeeze if there aren't any other real use-cases for intercepting/manipulating terminal output
